library/src/main/java/com/datatorrent/lib/streamquery/function/NvlFunction.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.lib.streamquery.function;
+
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.api.annotation.OutputPortFieldAnnotation;
+import com.datatorrent.common.util.BaseOperator;
+
+import java.util.ArrayList;
+
+/**
+ * An implementation of function NVL which lets you substitute a value when a null
+ * value is encountered.
+ *
+ * The syntax for NVL function is:
+ *      NVL (orig_string, replace_with)
+ *      orig_string: the string to test for a NULL value
+ *      replace_with: the value returned if orig_string is null.
+ *      (Note the function will return orig_string if it is not null)
+ *
+ * For example, with the following table
+ *
+ *      EMP table
+ *      ========
+ *      name   salary
+ *      ----------------
+ *      mike   1000
+ *      ben    null
+ *      dave   1500
+ *      SELECT NVL(salary, 0) FROM emp WHERE name='ben' will return 0.
+ *      SELECT NVL(salary, 0) FROM emp WHERE name='dave' will return 1500.
+ *
+ * @displayName NVL Function
+ * @category Stream Manipulators
+ * @tags sql nvl
+ * @since 0.3.4
+ */
+public class NvlFunction<T> extends BaseOperator
+{
+    /**
+     * Array to store column inputs during window.
+     */
+    private ArrayList<T> columns = new ArrayList<T>();
+
+    /**
+     * Array to store aliases input during window.
+     */
+    private ArrayList<T> aliases = new ArrayList<T>();
+
+    /**
+     * Number of input pairs processed in current window.
+     */
+    private int index = 0;
+
+    /**
+     * Helper to process a tuple
+     */
+    private void processInternal(T tuple, boolean isAlias) {
+        int size;
+        if (isAlias) {
+            aliases.add(tuple);
+            size = columns.size();
+        }
+        else {
+            columns.add(tuple);
+            size = aliases.size();
+        }
+
+        if (size > index) {
```

Comment:
I think we can simplify this a lot as follows

```
if(alias.size() > 0 && columns.size() > 0){
      emit(columns.remove(0), alias.remove(0))
}
```

There is no need of variable `index` and `size` and if assumption (that number of tuples received on both column and alias port will be same in a given window) holds true then no need to end window....